django/db/backends/mysql/introspection.py

```diff
@@ -289,7 +289,7 @@ def get_constraints(self, cursor, table_name):
                 constraints[index] = {
                     'columns': OrderedSet(),
                     'primary_key': False,
-                    'unique': False,
+                    'unique': not non_unique,
```

Comment:
This is also a separate issue, but it's not testable without functional constraints so we can leave it here.